### PR TITLE
Add dotnet library test

### PR DESCRIPTION
**Pull request by asvrada**
The dotnet binary is able to run in Mystikos, but hangs forever when running any tests.

Sample output in Mystikos
```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
^CMakefile:19: recipe for target 'run' failed
make: *** [run] Interrupt
```

Sample output outside Mystikos
```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:    97, Skipped:     0, Total:    97, Duration: 64 ms - /dotnet-lib-release/System.ValueTuple.Tests/net5.0-Release/System.ValueTuple.Tests.dll (net5.0)
```

Upon checking strace, there seems to be repeating pattern of following sys call

```
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
=== SYS_sched_yield(): pid=111 tid=119
    SYS_sched_yield(): return=0(0): pid=111 tid=119
...
=== SYS_mmap(addr=7fd109820000 length=65536(10000) prot=0 flags=34 fd=-1 offset=0): pid=111 tid=111
    SYS_mmap(): return=140534386806784(7fd0b6345000): pid=111 tid=111
=== SYS_munmap(addr=7fd0b6345000 length=65536(10000)): pid=111 tid=111
    SYS_munmap(): return=0(0): pid=111 tid=111
=== SYS_mmap(addr=7fd109830000 length=65536(10000) prot=0 flags=34 fd=-1 offset=0): pid=111 tid=111
    SYS_mmap(): return=140534386806784(7fd0b6345000): pid=111 tid=111
=== SYS_munmap(addr=7fd0b6345000 length=65536(10000)): pid=111 tid=111
    SYS_munmap(): return=0(0): pid=111 tid=111
=== SYS_mmap(addr=7fd109840000 length=65536(10000) prot=0 flags=34 fd=-1 offset=0): pid=111 tid=111
    SYS_mmap(): return=140534386806784(7fd0b6345000): pid=111 tid=111
=== SYS_munmap(addr=7fd0b6345000 length=65536(10000)): pid=111 tid=111
    SYS_munmap(): return=0(0): pid=111 tid=111
=== SYS_mmap(addr=7fd109850000 length=65536(10000) prot=0 flags=34 fd=-1 offset=0): pid=111 tid=111
    SYS_mmap(): return=140534386806784(7fd0b6345000): pid=111 tid=111
=== SYS_munmap(addr=7fd0b6345000 length=65536(10000)): pid=111 tid=111
    SYS_munmap(): return=0(0): pid=111 tid=111
```

Full strace file here: https://microsoft-my.sharepoint.com/:t:/p/zijiewu/EQvd6necQWpGvv_S3or9-VQBvoZcX-Bz5j0yy56eoqBfnA?e=XJDeFN

### CODE REVIEW

**Comment by asvrada** [comment]
Blocked for now, waiting for fix to Mystikos to unblock the testing

**Review by anakrish — APPROVED** [review]
Squash commits before merging. LGTM

**Pull request merged**
